runtime/js/ieee_754.js

```diff
@@ -17,74 +17,35 @@
 // along with this program; if not, write to the Free Software
 // Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
-//Provides: jsoo_floor_log2
-var log2_ok = Math.log2 && Math.log2(1.1235582092889474e307) === 1020;
-function jsoo_floor_log2(x) {
-  if (log2_ok) return Math.floor(Math.log2(x));
-  var i = 0;
-  if (x === 0) return Number.NEGATIVE_INFINITY;
-  if (x >= 1) {
-    while (x >= 2) {
-      x /= 2;
-      i++;
-    }
-  } else {
-    while (x < 1) {
-      x *= 2;
-      i--;
-    }
-  }
-  return i;
-}
+//Provides: caml_bufferf64
+var caml_bufferf64 = new Float64Array(1);
+//Provides: caml_bufferi32
+//Requires: caml_bufferf64
+var caml_bufferi32 = new Int32Array(caml_bufferf64.buffer);
+//Provides: caml_bufferf32
+//Requires: caml_bufferf64
+var caml_bufferf32 = new Float32Array(caml_bufferf64.buffer);
 
 //Provides: caml_int64_bits_of_float const
-//Requires: jsoo_floor_log2, caml_int64_create_lo_mi_hi
+//Requires: caml_int64_create_lo_mi_hi
+//Requires: caml_bufferf64
+//Requires: caml_bufferi32
 function caml_int64_bits_of_float(x) {
-  if (!Number.isFinite(x)) {
-    if (Number.isNaN(x)) return caml_int64_create_lo_mi_hi(1, 0, 0x7ff0);
-    if (x > 0) return caml_int64_create_lo_mi_hi(0, 0, 0x7ff0);
-    else return caml_int64_create_lo_mi_hi(0, 0, 0xfff0);
-  }
-  var sign =
-    x === 0 && 1 / x === Number.NEGATIVE_INFINITY
-      ? 0x8000
-      : x >= 0
-        ? 0
-        : 0x8000;
-  if (sign) x = -x;
-  // Int64.bits_of_float 1.1235582092889474E+307 = 0x7fb0000000000000L
-  // using Math.LOG2E*Math.log(x) in place of Math.log2 result in precision lost
-  var exp = jsoo_floor_log2(x) + 1023;
-  if (exp <= 0) {
-    exp = 0;
-    x /= Math.pow(2, -1026);
-  } else {
-    x /= Math.pow(2, exp - 1027);
-    if (x < 16) {
-      x *= 2;
-      exp -= 1;
-    }
-    if (exp === 0) {
-      x /= 2;
-    }
-  }
-  var k = Math.pow(2, 24);
-  var r3 = x | 0;
-  x = (x - r3) * k;
-  var r2 = x | 0;
-  x = (x - r2) * k;
-  var r1 = x | 0;
-  r3 = (r3 & 0xf) | sign | (exp << 4);
+  caml_bufferf64[0] = x;
+  var lo32 = caml_bufferi32[0];
+  var hi32 = caml_bufferi32[1];
```

Comment:
This assumes a little endian architecture. It would be more portable to use a DataView:
```js
  var buffer = new ArrayBuffer(8);
  var view = new DataView(buffer);
  view.setFloat64(0, x, true); // little endian
  var lo32 = view.getUint32(0, true);
  var hi32 = view.getUint32(4, true);
```